source/containers/document-pii-detection/main.py

```diff
@@ -0,0 +1,208 @@
+        
+        parser = ParserFactory.create_parser(file_type=file_type, s3_client=s3_client)
+
+        for sample_file in sample_files:
+            object_key = f"{file_path}/{sample_file}{file_type}"
+            file_content = parser.load_content(bucket_name, object_key)
+            file_contents[f"{sample_file}"] = file_content
+
+    elif file_category == 'include_files':
+        for sample_file in sample_files:
+            file_contents[f"{sample_file}"] = ['This file is marked as Contains-PII.']
+    
+    elif file_category == 'exclude_files':
+        for sample_file in sample_files:
+            file_contents[f"{sample_file}"] = ['This file is marked as Non-PII.']
+            
+    return file_contents
+
+def process_file(parser, bucket_name, object_key):
+    """
+    Process a single file.
+    """
+    file_content = parser.load_content(bucket_name, object_key)
+
+    json_format_content = {}
+    json_format_content[f"{object_key}"] = file_content
+    
+    return json_format_content
+
+def create_glue_table(glue_client, database_name, table_name, glue_table_info):
+
+    # Check if table exists
+    try:
+        response = glue_client.get_table(
+            DatabaseName=database_name,
+            Name=table_name
+        )
+        print(f"Table '{table_name}' exists in database '{database_name}'. Updating table...")
+        response = glue_client.update_table(
+            DatabaseName=database_name,
+            TableInput=glue_table_info
+        )
+    except glue_client.exceptions.EntityNotFoundException:
+        print(f"Table '{table_name}' does not exist in database '{database_name}'. Creating table...")
+        response = glue_client.create_table(
+            DatabaseName=database_name,
+            TableInput=glue_table_info
+        )
+
+    print(response)
+
+def main(param_dict):
+    original_bucket_name = param_dict['SourceBucketName']
+    crawler_result_bucket_name = param_dict['ResultBucketName']
+    region_name = param_dict['RegionName']
+
+    crawler_result_object_key = f"crawler_results/{original_bucket_name}_info.json"
+    destination_database = f"SDPS-unstructured-{original_bucket_name}"
+
+    s3_client = boto3.client('s3', region_name = region_name)
+    glue_client = boto3.client('glue', region_name = region_name)
+
+    # 1. Create a Glue Database
+    try:
+        response = glue_client.create_database(
+            DatabaseInput={
+                'Name': destination_database
+            }
+        )
+    except glue_client.exceptions.AlreadyExistsException:
+        print(f"Database '{destination_database}' already exists. Skipping database creation...")
+
+    # 2. Download the crawler result from S3 and 
+    with tempfile.NamedTemporaryFile(mode='w') as temp:
+        temp_file_path = temp.name
+        s3_client.download_file(Bucket=crawler_result_bucket_name, Key=crawler_result_object_key, Filename=temp_file_path)
+        bucket_info = json.load(open(temp_file_path, 'r'))
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code might contain a resource leak. Resource leaks can cause your system to slow down or crash.

**Fix**
Consider closing the resource returned by the following method call: *open*. The resource is allocated by call *builtins.open*. Execution paths that do not contain closure statements were detected. To prevent this resource leak, close the object returned by *open()* in a try-finally block or declare it using a `with` statement.

**More info**
[View details about the `with` statement in the Python developer&#39;s guide](https://www.python.org/dev/peps/pep-0343/) (external link).

Reply:
Resolve by using `with open()` to load file content.